pypsa/io.py

```diff
@@ -992,106 +1073,122 @@ def import_components_from_dataframe(
             )
 
     non_static_attrs_in_df = non_static_attrs.index.intersection(dataframe.columns)
-    old_df = network.df(cls_name)
+    old_df = n.df(cls_name)
     new_df = dataframe.drop(non_static_attrs_in_df, axis=1)
+
+    # Handle duplicates
+    duplicated_components = old_df.index.intersection(new_df.index)
+    if len(duplicated_components) > 0:
+        if not overwrite:
+            logger.warning(
+                "The following %s are already defined and will be skipped "
+                "(use overwrite=True to overwrite): %s",
+                n.components[cls_name]["list_name"],
+                ", ".join(duplicated_components),
+            )
```

Comment:
I think we should raise an error here instead. Like this, we give a clear feedback, logger warnings are often ignored by users according to my experience.